### PR TITLE
Add color aspect support in video encoder

### PR DESCRIPTION
**Pull request by TianmiChen**
android.mediav2.cts.EncoderColorAspectsTest#testColorAspects

Tracked-On: OAM-102737
Signed-off-by: Chen, Tianmi <tianmi.chen@intel.com>

### CODE REVIEW

**Comment by TianmiChen** [comment]
@dyang23 , @zhangyichix , @huangyuanjun, please take a review.

Thanks,

File: c2_components/include/mfx_c2_encoder_component.h [hunk 1]
```diff
@@ -202,6 +210,10 @@ class MfxC2EncoderComponent : public MfxC2Component
     MfxC2VppWrapp m_vpp;
     MfxC2Conversion m_inputVppType;
 
+    mfxExtVideoSignalInfo m_signalInfo;
+    std::shared_ptr<C2StreamColorAspectsInfo::input> mColorAspects;
```
Comment (zhangyichix) [inline c2_components/include/mfx_c2_encoder_component.h:214]: Please follow the naming style like m_xxx

Comment (zhangyichix) [inline c2_components/include/mfx_c2_encoder_component.h:214]: Please follow the naming style like m_xxx

**Review by zhangyichix — CHANGES_REQUESTED** [review]

File: c2_components/src/mfx_c2_encoder_component.cpp [hunk 2]
```diff
@@ -580,14 +606,31 @@ mfxStatus MfxC2EncoderComponent::ResetSettings()
...
     m_mfxVideoParamsConfig.ExtParam = nullptr;
 
-    if (m_encoderType == ENCODER_VP9) {
+    MFX_DEBUG_TRACE__mfxStatus(mfx_res);
+    return mfx_res;
+}
+
+void MfxC2EncoderComponent::AttachExtBuffer()
+{
+    MFX_DEBUG_TRACE_FUNC;
```
Comment (dyang23) [inline c2_components/src/mfx_c2_encoder_component.cpp:615]: Whether the h264/h265 different with VP9, is the msdk API requirement?

**Review by dyang23 — COMMENTED** [review]
One question inlined.

File: c2_components/src/mfx_c2_encoder_component.cpp (hunk 2 above)
Comment (TianmiChen) [reply c2_components/src/mfx_c2_encoder_component.cpp:615]: Video Signal Info is applicable to h264 and h265 per codec standard. VP9 bitstream doesn't contain video signal information.

File: c2_components/include/mfx_c2_encoder_component.h (hunk 1 above)
Comment (TianmiChen) [reply c2_components/include/mfx_c2_encoder_component.h]: Updated.

**Review by dyang23 — CHANGES_REQUESTED** [review]
LGTM， thanks

**Review by dyang23 — APPROVED** [review]
LGTM， Thanks

**Review by zhangyichix — APPROVED** [review]

**Review by YuanjunHuang — APPROVED** [review]

**Pull request merged**